modules/ingester/flush.go

```diff
@@ -286,11 +286,13 @@ func (i *Ingester) handleComplete(ctx context.Context, op *flushOp) (retry bool,
 
 	// add a flushOp for the block we just completed
 	// No delay
-	i.enqueue(&flushOp{
-		kind:    opKindFlush,
-		userID:  instance.instanceID,
-		blockID: op.blockID,
-	}, false)
+	if i.cfg.FlushObjectStorage {
```

Comment:
We need to mark blocks flushed or change some logic so complete blocks are cleared at some point.